src/runtime_src/core/tools/xbmgmt2/flash/xgq.h

```diff
@@ -0,0 +1,34 @@
+/**
+ * Copyright (C) 2021 Xilinx, Inc
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"). You may
+ * not use this file except in compliance with the License. A copy of the
+ * License is located at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+#ifndef _XGQ_FLASHER_H_
+#define _XGQ_FLASHER_H_
+
+#include <iostream>
+#include "core/common/system.h"
+#include "core/common/device.h"
+
+class XGQ_Flasher
+{
+public:
+    XGQ_Flasher(std::shared_ptr<xrt_core::device> dev);
+    int xclUpgradeFirmware(std::istream& binStream);
+
+private:
+    std::shared_ptr<xrt_core::device> m_device;
+};
+

```

Comment:
Please don't add classes or structs and public member functions without inline documentation about purpose of class and what the functions do (arguments and return).

Reply:
This is clone of xospiversal.h, I will discuss with @AShivangi and fix this.